pkg/stanza/fileconsumer/matcher/matcher.go

```diff
@@ -66,11 +71,21 @@ func New(c Criteria) (*Matcher, error) {
 		return nil, fmt.Errorf("exclude: %w", err)
 	}
 
+	matcher := &Matcher{
```

Comment:
We are instantiating this here but also re-instantiating at the end of the function. Previously this worked because we were returning the struct immediately. Let's just update the rest of the function to modify this struct instead of recreating it.

Reply:
Fixed in 747cfac374.